### PR TITLE
feat: create form builder mobile view

### PR DESCRIPTION
**Pull request by mantariksh**
_Depends on #3692_

Implements the form field creation and editing flow on mobile.

**Note that drag and drop has not yet been implemented**. This is due to the complexity of having to close the add field modal when a field is dragged, but still keeping it in the DOM as required by `react-beautiful-dnd`.

### Create field
mobile-builder-create

### Edit field
mobile-builder-edit

### CODE REVIEW

File: frontend/src/features/admin-form/create/builder-and-design/BuilderAndDesignDrawer/EditFieldDrawer/edit-fieldtype/common/useEditFieldForm.ts [hunk 1]
```diff
@@ -51,25 +95,63 @@ export const useEditFieldForm = <FormShape, FieldShape extends FieldBase>({
...
+  const handleUpdateField = editForm.handleSubmit((inputs) => {
+    const updatedFormField = transform.output(inputs, field)
+    if (stateData.state === BuildFieldState.CreatingField) {
+      return createFieldMutation.mutate(updatedFormField, {
+        onSuccess: onSaveSuccess,
+      })
+    } else if (stateData.state === BuildFieldState.EditingField) {
+      return editFieldMutation.mutate(
+        // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+        // @ts-ignore
```
Comment (timotheeg) [inline frontend/src/features/admin-form/create/builder-and-design/BuilderAndDesignDrawer/EditFieldDrawer/edit-fieldtype/common/useEditFieldForm.ts:121]: Casting wouldn't work here like it's done below for `updateEditState()`?

File: frontend/src/features/admin-form/create/builder-and-design/BuilderAndDesignDrawer/EditFieldDrawer/EditFieldDrawer.tsx [hunk 2]
```diff
@@ -150,30 +106,27 @@ export const EditFieldDrawer = (): JSX.Element | null => {
...
-    case BasicField.Radio:
-      return <EditRadio {...props} field={field} />
-    default:
-      return <div>TODO: Insert field options here</div>
-  }
-})
+export const MemoFieldDrawerContent = memo(
+  (props: MemoFieldDrawerContentProps) => {
+    // Extract field variable just to get field.fieldType types to cooperate
+    const field = useMemo(() => props.field, [props.field])
```
Comment (timotheeg) [inline frontend/src/features/admin-form/create/builder-and-design/BuilderAndDesignDrawer/EditFieldDrawer/EditFieldDrawer.tsx:114]: Not from this PR, but... this useMemo()... 🙄 😑

**Review by karrui — APPROVED** [review]

File: frontend/src/features/admin-form/create/builder-and-design/BuilderAndDesignContent/BuilderFields.tsx [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+interface BuilderFieldsProps {
+  fields: AdminFormDto['form_fields']
+  isDraggingOver: boolean
+}
+
+export const BuilderFields = memo(
+  ({ fields, isDraggingOver }: BuilderFieldsProps) => {
+    const questionNumbers = useMemo(
+      () => getBuilderQuestionNumbers(fields),
+      [fields],
```
[the remark this answers is not in the document]
Comment (mantariksh) [reply frontend/src/features/admin-form/create/builder-and-design/BuilderAndDesignContent/BuilderFields.tsx:18]: removed the `if` block in c65c0cbd as `fields` cannot be falsy. `BuilderAndDesignContent` doesn't render `BuilderFields` if `fields` is falsy.

File: frontend/src/features/admin-form/create/builder-and-design/BuilderAndDesignDrawer/EditFieldDrawer/edit-fieldtype/common/useEditFieldForm.ts (hunk 1 above)
Comment (mantariksh) [reply frontend/src/features/admin-form/create/builder-and-design/BuilderAndDesignDrawer/EditFieldDrawer/edit-fieldtype/common/useEditFieldForm.ts]: tried it and it worked, thanks! eb40295a

File: frontend/src/features/admin-form/create/builder-and-design/BuilderAndDesignDrawer/EditFieldDrawer/EditFieldDrawer.tsx (hunk 2 above)
Comment (mantariksh) [reply frontend/src/features/admin-form/create/builder-and-design/BuilderAndDesignDrawer/EditFieldDrawer/EditFieldDrawer.tsx]: removed the `useMemo` in 11533f48 after reading [this article](https://kentcdodds.com/blog/usememo-and-usecallback)!

**Review by timotheeg — APPROVED** [review]

File: frontend/src/features/admin-form/create/builder-and-design/BuilderAndDesignDrawer/EditFieldDrawer/EditFieldDrawer.tsx (hunk 2 above)
Comment (timotheeg) [reply frontend/src/features/admin-form/create/builder-and-design/BuilderAndDesignDrawer/EditFieldDrawer/EditFieldDrawer.tsx]: Thanks for that 🙏 !

I'm scheduling an item for a short eng discussion on this topic tomorrow.

**Pull request merged**